### PR TITLE
Update to Dynamo Core 2.8

### PR DESCRIPTION
**Pull request by mmisol**
### Purpose

Includes:
- Dynamo NuGet update
- Preloading the Dynamo CPython3 engine
- Revit customizations for Dynamo CPython3 engine

Known caveats:
- No tests for new Python engine
- Dynamo Core is no longer CLS-compliant creating hundreds of warnings
- Dynamo NuGets for 2.8 are still in beta

### Declarations

Check these if you believe they are true

- [ ] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] Snapshot of UI changes, if any.

### FYIs

@mjkkirschner  @QilongTang

### CODE REVIEW

**Comment by mjkkirschner** [comment]
Any idea when / how we lost CLS compliance? Not that this is particularly important IMO... but we may want to silence those.

**Comment by mmisol** [comment]
Yes, it was here https://github.com/DynamoDS/Dynamo/pull/10668. I was told by @QilongTang this was resolved as a way to silence SQ warnings. I was trying to see if we had other ways of dealing with this, maybe silencing the warnings in SQ itself or something like that.

**Comment by QilongTang** [comment]
@mjkkirschner Was discussed in slack when first time we tackle SQ warnings, according to SQ we are not really following CLS compliant rules so less meaningful to keep that setting. Let me know.

**Comment by mjkkirschner** [comment]
oh, I see, perhaps it would be appropriate to add a pragma to DynamoRevit disabling the warnings if we have no intention of being CLS compliant - I know in some cases we can't - because we can't change API names at this time.

File: src/DynamoRevit/DynamoRevit.csproj [hunk 1]
```diff
@@ -93,7 +97,11 @@
       <RequiredTargetFramework>3.0</RequiredTargetFramework>
     </Reference>
     <Reference Include="ProtoGeometry">
-      <HintPath>$(PACKAGESPATH)\DynamoVisualProgramming.ZeroTouchLibrary\lib\net47\ProtoGeometry.dll</HintPath>
+      <HintPath>$(PACKAGESPATH)\DynamoVisualProgramming.ZeroTouchLibrary\lib\net48\ProtoGeometry.dll</HintPath>
+      <Private>False</Private>
+    </Reference>
+    <Reference Include="Python.Runtime">
+      <HintPath>$(PACKAGESPATH)\pythonnet\lib\net40\Python.Runtime.dll</HintPath>
```
Comment (mjkkirschner) [inline src/DynamoRevit/DynamoRevit.csproj:104]: what was this actually needed for?

Comment (mmisol) [reply src/DynamoRevit/DynamoRevit.csproj:104]: A reference to `PyScope` is required when adding the `UnwrapElement` function.

**Comment by ZiyunShang** [comment]
Hi @mjkkirschner @QilongTang , @AndyDu1985 is on vacation now, and I should confirm whether we want to upgrade to DynamoCore 2.8.0 now and will be released next version.

**Comment by mmisol** [comment]
@ZiyunShang Sorry if I'm misunderstanding. The planned work we are currently executing is to get 2.8 to be the version of Dynamo for global launch. Are you saying this is still being discussed or that you are not informed of any decision yet? Thanks!
Copying @Amoursol our product manager to keep him posted.

**Comment by QilongTang** [comment]
@ZiyunShang Is there a way we can get Dynamo 2.8 on Revit master branch? I think this PR is the suggested work

File: src/DynamoRevit/Models/RevitDynamoModel.cs [hunk 2]
```diff
@@ -389,18 +390,31 @@ private void SetupPython()
...
+            CPythonEvaluator.OutputMarshaler.RegisterMarshaler(
+                (Element element) => element.ToDSType(true));
+
+            // Turn off element binding during cpython script execution
+            CPythonEvaluator.EvaluationBegin +=
+                (a, b, c, d) => ElementBinder.IsEnabled = false;
+            CPythonEvaluator.EvaluationEnd += (a, b, c, d) => ElementBinder.IsEnabled = true;
+
+            // register UnwrapElement method in cpython
+            CPythonEvaluator.EvaluationBegin += (a, scope, c, d) =>
```
Comment (mjkkirschner) [inline src/DynamoRevit/Models/RevitDynamoModel.cs:413]: should we be unsubscribing these when python nodes are removed from the workspace?

Comment (mmisol) [reply src/DynamoRevit/Models/RevitDynamoModel.cs:413]: I didn't get that @mjkkirschner . These events are static, not really associated to a Python node in particular.

Comment (mjkkirschner) [reply src/DynamoRevit/Models/RevitDynamoModel.cs:413]: whoops! my mistake.

**Comment by QilongTang** [comment]
@AndyDu1985 @ZiyunShang Is it fine to merge now or you would rather wait for the final RC2.8.0 to be out by end of month to do a single sweep?

**Comment by AndyDu1985** [comment]
> @AndyDu1985 @ZiyunShang Is it fine to merge now or you would rather wait for the final RC2.8.0 to be out by end of month to do a single sweep?

It's OK have this version in DynamoRevit.

**Pull request merged**